src/components/Button/Button.tsx

```diff
@@ -44,8 +45,16 @@ export class Button extends React.Component<IButtonProps, IButtonState> {
                                                // if not utilize default button disabled behavior.
     });
 
+    const iconClassName = css('ms-Icon', iconTypeClass, {
```

Comment:
we should really use more semantic meanings here. 

e.g.:

A primary button's icon should be white:

.ms-Button.ms-Button--primary .ms-Icon {
color: @ms-color-white;
}